RACK-Ontology/ontology/REQUIREMENTS.sadl

```diff
@@ -10,62 +10,73 @@
  * of the Defense Advanced Research Projects Agency (DARPA).
  */
 
+/************** edit history *****************
+  *
+  * dwa - 11 Nov 2020 - reduced cardinality of several
+  * dwa - 11 Nov 2020 - removed incorrect associations
+  * dwa - 11 Nov 2020 - removed unnecessary subclassing
+  *
+  *********************************************/
+
 uri "http://arcos.rack/REQUIREMENTS" alias Rq.
-import "http://arcos.rack/PROV-S".
+import "http://arcos.rack/HAZARD".
 
 REQUIREMENT
 	(note "Captures (both high- and low-level) properties of a process or artifact that are to be assessed")
 	is a type of ENTITY.
 
-	text (note "Description of the REQUIREMENT") describes REQUIREMENT with values of type string.
+	text (note "Description of the REQUIREMENT") describes REQUIREMENT with a single value of type string.
 	// The "text" string can optionally be subdivided into sub-parts : givenText, ifText, thenText
 	// This corresponds to how rules are defined in SADL and note that in a SADL rule whatever is in
 	// givenText can simply be included in the ifText part.
 	// Note that modulo naming changes this structure also matches the structure
 	// that GrammaTech had which was : Given, When, Then parts.
 	// The structure of requirements in ASSERT subsume this structure of requirements. 
 	givenText (note "optional part that can be used to set certain conditions apart from other conditions in the ifText part") describes REQUIREMENT with values of type string.
+	givenText describes REQUIREMENT with at most 1 value.
 	ifText (note "conditions under which this requirement holds") describes REQUIREMENT with values of type string.
+	ifText describes REQUIREMENT with at most 1 value.
```

Comment:
I don't think this should have a cardinality restriction. there is no reason that you could not have multiple. 

Also in a common should the there be a new ENTITY defined for GIVEN_TEXT, in general it seems like the ravage should not be a string for anything outside the THING class. since how would TA3 be able to use/query for that. Seems like we should use a specific class, or this should be moved to the supplemental ontology

Reply:
I'm fine with relaxing the cardinality restriction. Problem I see is that it's not clear whether the ifTexts are conjunctive or disjunctive if we do that. Is it for example that ALL of the ifs apply? or that ANY of the ifs apply? If Dan has a clean answer for that question, we can probably close this one.